services/cptec/weather.js

```diff
@@ -99,6 +99,32 @@ export const getCurrentAirportWeather = async (icaoCode) => {
   return [];
 };
 
+/**
+ * Get weather predicion for the next 7 days with location based
+ * @param {float} lat
+ * @param {float} long
+ * @returns {object}
+ */
+export const getPredictionWeatherByLocation = async (lat, long) => {
+  const url = `${CPTEC_URL}/cidade/7dias/${lat}/${long}/previsaoLatLon.xml`;
+
+  const weatherPredictions = await axios.get(url, {
+    responseType: 'application/xml',
+    responseEncoding: 'binary',
+  });
+
+  const parsed = parser.parse(weatherPredictions.data);
+
+  if (parsed.cidade) {
+    const jsonData = formatPrediction(parsed);
+    if (jsonData.cidade === 'null') {
+      return null;
+    }
+    return jsonData;
+  }
+  return [];
```

Comment:
O que acha dessa mudança? O que muda aqui é que foi adotada a estratégia Return Early Pattern e em vez de retornar [] quando não tem cidade, é retornado null

```suggestion
  if (!parsed.cidade) {
    return null;
  }
  
  const jsonData = formatPrediction(parsed);
  if (jsonData.cidade === 'null') {
    return null;
  }
  return jsonData;
```